src/app/core/guards/server-side-rendering.guard.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+
+/**
+ * guards a route against server side rendering (e.g. if the logic requires information only available in browser rendering)
+ */
+@Injectable({ providedIn: 'root' })
+export class ServerSideRenderingGuard implements CanActivate {
```

Comment:
How about renaming it to `NoServerSideRenderingGuard`, that is more like what it does.

Reply:
@shauke My train of thought:
"In computer programming, a guard is a boolean expression that must evaluate to true if the program execution is to continue in the branch in question." [[Wiki]](https://en.wikipedia.org/wiki/Guard_(computer_science))

So the guard does not "guard against" SSR, but instead the condition "no SSR" must be true to activate that route.

Reply:
The renaming sounds good/reasonable to me.